### PR TITLE
Show Faust Toolbar based on user preferences in WordPress

### PR DESCRIPTION
**Pull request by blakewilson**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.

## Description

This PR shows or hides the Faust Toolbar based on the user's toolbar preferences in WordPress:
Screenshot 2023-03-31 at 11 13 15 AM

A field was added to the `viewer` model in WPGraphQL called `shouldShowFaustToolbar` which we use to either show or hide the toolbar component.

If the user is using a version of Faust where this field does not exist, the query will throw an error, and show the toolbar. If the query fails, there is no blocking errors, the behavior just falls back to showing the toolbar.

## Testing

Tests have been provided. Additionally, here are some manual testing instructions:

1. Checkout this branch
2. Build the core and cli packages
3. In the example app, ensure `experimentalToolbar` is set
4. Make sure you are using the version of FaustWP from this branch
5. Start the dev server
6. Authenticate via previews
7. Notice the toolbar will now show based on your preferences in WordPress

### CODE REVIEW

**Comment by josephfusco** [comment]
Tracking https://github.com/wp-graphql/wp-graphql/pull/2763

File: packages/faustwp-core/src/components/Toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -183,3 +172,54 @@ export function Toolbar({ seedNode }: ToolbarProps) {
...
+ * Renders a Toolbar that is based on WordPress' own toolbar.
+ */
+export function Toolbar({ seedNode }: ToolbarProps) {
+  const { isAuthenticated } = useAuth();
+
+  if (isAuthenticated !== true) {
+    return null;
+  }
+
+  return <ToolbarAwaitUser seedNode={seedNode} />;
```
Comment (matthewguywright) [inline packages/faustwp-core/src/components/Toolbar/Toolbar.tsx:224]: That's slick

**Review by matthewguywright — APPROVED** [review]

File: plugins/faustwp/includes/graphql/callbacks.php [hunk 2]
```diff
@@ -44,6 +44,34 @@ function register_templates_field() {
...
+			'resolve' => function() {
+				$user                    = wp_get_current_user();
+				$toolbar_preference_meta = get_user_meta( $user->ID, 'show_admin_bar_front', true );
+
+				return 'true' === $toolbar_preference_meta ? true : false;
+			},
+		)
+	);
+}
+
```
Comment (josephfusco) [inline plugins/faustwp/includes/graphql/callbacks.php:74]: 💯

**Review by josephfusco — APPROVED** [review]
Nice work on this @blakewilson! LGTM!

**Pull request merged**